eval.go

```diff
@@ -237,14 +237,28 @@ nextFileMatch:
 		}
 
 		for ; nextDoc < docCount; nextDoc++ {
+			repoID := d.repos[nextDoc]
+			repoMetadata := d.repoMetaData[repoID]
+
 			// Skip tombstoned docs
-			if d.repoMetaData[d.repos[nextDoc]].Tombstone {
+			if repoMetadata.Tombstone {
 				continue
 			}
 
+			if opts.EvaluateFileTombstones {
+
+				if repoMetadata.FileTombstones != nil {
```

Comment:
Having to manage a possibly `nil` FileTombstones map also strikes me as awkward, and maybe we should move to the sorted list of document IDs approach sooner rather than later. (The other ways that I can think about this are to have nil guards everywhere, or to mandate the use of a factory `zoket.NewRepository()` that takes care of initializing the field for you - neither of these seem great).

Reply:
you can do lookups on nil maps. This works and `ok` is false

```go
var m map[string]struct{}
m = nil
_, ok := m["foo"]
```

that should simplify this control flow a bit.

Also you can skip map lookups if `len(...)` is zero. Benefit of this is you skip the more expensive fileName creation work.